components/automate-ui/src/app/modules/infra-proxy/edit-infra-role-modal/edit-infra-role-modal.component.ts

```diff
@@ -0,0 +1,236 @@
+        IdMapper.transform(this.defaultAttributeForm.controls.default.value.trim()));
+    }
+  }
+
+  onChangeDefaultJson(event: { target: { value: string } } ) {
+    const newValue = event.target.value;
+    try {
+      JSON.parse(newValue);
+      this.defaultAttrParseError = false;
+    } catch (ex) {
+      this.defaultAttrParseError = true;
+    }
+  }
+
+  onChangeOverrideJson(event: { target: { value: string } } ) {
+    const newValue = event.target.value;
+    try {
+      JSON.parse(newValue);
+      this.overrideAttrParseError = false;
+    } catch (ex) {
+      this.overrideAttrParseError = true;
+    }
+  }
+
+  updateInfrarole(): void {
+    this.updateSuccessful = false;
+    this.updateInProgress = true;
+
+    let role: InfraRole = {
+      server_id: this.serverId,
+      org_id: this.orgId,
+      name: this.role.name,
+      description: this.role.description,
+
+      // these are filled in by the switch below
+      run_list: [],
+      default_attributes: '',
+      override_attributes: ''
+    };
+
+    switch (this.label) {
+      case 'Run List':
+        role = { ...role,
+          run_list: this.isRunlist ? this.selectedRunLists : this.role.run_list,
+          default_attributes: JSON.parse(this.role.default_attributes),
+          override_attributes: JSON.parse(this.role.override_attributes)
+        };
+        break;
+
+      case 'Default':
+        role = { ...role,
+          run_list: this.role.run_list,
+          default_attributes: JSON.parse(
+            this.defaultAttributeForm.controls['default'].value.replace(/\r?\n|\r/g, '')),
+          override_attributes: JSON.parse(this.role.override_attributes)
+        };
+        break;
+
+      case 'Override':
+        role = { ...role,
+          run_list: this.selectedRunLists.length > 0 ? this.selectedRunLists : this.role.run_list,
+          default_attributes: JSON.parse(this.role.default_attributes),
+          override_attributes: JSON.parse(
+            this.overrideAttributeForm.controls['override'].value.replace(/\r?\n|\r/g, ''))
+        };
+        break;
+    }
+    this.updatingData(role);
+  }
+
+  private resetEditModal(): void {
+    this.creating = false;
+    this.defaultAttributeForm.markAsPristine();
+    if (this.label === 'Default') {
+      this.defaultAttributeForm.controls.default.setValue(this.jsonText);
+    }
+    if (this.label === 'Override') {
+      this.overrideAttributeForm.controls.override.setValue(this.jsonText);
+    }
+    this.overrideAttributeForm.markAsPristine();
+    this.selectedRunLists = [];
+    this.defaultAttrParseError = false;
+    this.overrideAttrParseError = false;
+    this.showbutton = false;
+    this.conflictErrorEvent.emit(false);
+  }
+
+  private updatingData(role: InfraRole) {
+    this.store.dispatch(new UpdateRole(role));
+  }
+
+  private isNavigationKey(event: KeyboardEvent): boolean {
+    return event.key === 'Shift' || event.key === 'Tab';
+  }
+}
```

Comment:
Same comment as above, we can now import this from the utilities folder.

Reply:
updated the changes